tests/pipeline/test_multidimensional_dataset.py

```diff
@@ -0,0 +1,234 @@
+                MD.slice(*args, **kwargs)
+
+        # insufficient positional
+        expect_slice_fails(
+            expected_msg=(
+                'no label provided for the following axes: axis_0, axis_1'
+            ),
+        )
+        expect_slice_fails(
+            'a',
+            expected_msg='no label provided for the following axis: axis_1',
+        )
+
+        # too many positional
+        expect_slice_fails(
+            'a', 'b', 'c',
+            expected_msg='MD has 2 free axes but 3 were given',
+        )
+
+        # mismatched keys
+        expect_slice_fails(
+            axis_2='??',
+            expected_msg='MD does not have the following axis: axis_2',
+        )
+        expect_slice_fails(
+            axis_1='??', axis_2='??',
+            expected_msg='MD does not have the following axis: axis_2',
+        )
+        expect_slice_fails(
+            axis_0='??', axis_1='??', axis_2='??',
+            expected_msg='MD does not have the following axis: axis_2',
+        )
+
+        # the extra keyword axes should be sorted
+        expect_slice_fails(
+            axis_3='??', axis_2='??',
+            expected_msg='MD does not have the following axes: axis_2, axis_3',
+        )
+
+    def test_slice_unknown_axis_label(self):
+        class MD(MultiDimensionalDataSet):
+            free_axes = [
+                ('axis_0', {'a', 'b', 'c'}),
+                ('axis_1', {'c', 'd', 'e'}),
+            ]
+
+        def expect_slice_fails(*args, **kwargs):
+            expected_msg = kwargs.pop('expected_msg')
+
+            with assert_raises_str(ValueError, expected_msg):
+                MD.slice(*args, **kwargs)
+
+        expect_slice_fails(
+            'not-in-0', 'c',
+            expected_msg="'not-in-0' is not a value along the axis_0 axis",
+        )
+        expect_slice_fails(
+            axis_0='not-in-0', axis_1='c',
+            expected_msg="'not-in-0' is not a value along the axis_0 axis",
+        )
+
+        expect_slice_fails(
+            'a', 'not-in-1',
+            expected_msg="'not-in-1' is not a value along the axis_1 axis",
+        )
+        expect_slice_fails(
+            axis_0='a', axis_1='not-in-1',
+            expected_msg="'not-in-1' is not a value along the axis_1 axis",
+        )
+
+    def test_inheritence(self):
+        class Parent(MultiDimensionalDataSet):
+            free_axes = [
+                ('axis_0', {'a', 'b', 'c'}),
+                ('axis_1', {'d', 'e', 'f'}),
+            ]
+
+            column_0 = Column('f8')
+            column_1 = Column('?')
+
+        class Child(Parent):
+            column_2 = Column('O')
+            column_3 = Column('i8', -1)
+
+        assert_is_subclass(Child, Parent)
+        assert_equal(Child.free_axes, Parent.free_axes)
+
+        ParentSlice = Parent.slice(axis_0='a', axis_1='d')
+        ChildSlice = Child.slice(axis_0='a', axis_1='d')
+
+        assert_is_not_subclass(ChildSlice, ParentSlice)
```

Comment:
Is there a reason we specifically don't want these to be subclasses? I was actually originally expecting them to be, though I'm not sure how to achieve that offhand. I was just surprised that we explicitly added a test to make sure that they **weren't** subclasses.